chromadb/db/impl/grpc/client.py

```diff
@@ -288,13 +288,18 @@ def create_collection(
 
     @overrides
     def delete_collection(
-        self, id: UUID, tenant: str = DEFAULT_TENANT, database: str = DEFAULT_DATABASE
+        self,
+        id: UUID,
+        tenant: str = DEFAULT_TENANT,
+        database: str = DEFAULT_DATABASE,
+        segments: Optional[Sequence[UUID]] = None,
```

Comment:
It feels odd to me that our design here pushes the responsibility of getting the segments to the caller. I think it should be possible to transparently have lower layers do this since the association between collection and segment is tracked in the segments table?

Reply:
+1 I agree